scanner/scanner.go

```diff
@@ -495,6 +514,26 @@ func (s *Scanner) scanDoubleQuote(ctx *Context) (*token.Token, error) {
 	)
 }
 
+func (s *Scanner) validateDocumentSeparatorMarker(ctx *Context, src []rune) error {
+	if len(src) < 3 {
+		return nil
+	}
+	var marker string
+	if len(src) == 3 {
+		marker = string(src)
+	} else {
+		marker = strings.TrimRightFunc(string(src[:4]), func(r rune) bool {
+			return r == ' ' || r == '\t' || r == '\n' || r == '\r'
+		})
+	}
```

Comment:
Perhaps we can simplify:
```suggestion
	marker := string(src)
	if len(src) != 3 {
		marker = strings.TrimRightFunc(string(src[:4]), unicode.IsSpace)
	}
```